### ISSUE TITLE
JER doesn't show custom ores in JEI

### ISSUE THREAD

**Issue by lAimBotl** [RundownRhino/RegionScanner#16]
Modded ores that got scanned don't show up in "world gen" in JEI in game.

It also misses some modded ores like Immersive engineering ores in the worldgen file.

Versions:
- jei-1.20.1-forge-15.14.0.64
- emi-1.1.12+1.20.1+forge
- JustEnoughResources-1.20.1-1.4.0.247
- Region scanner v0.3.5
- Mc 1.20.1
- Forge: 47.3.7

World Gen .json File (put in /config folder of modpack)
[world-gen.json](https://github.com/user-attachments/files/16851849/world-gen.json)

**Comment by RundownRhino** [comment]
You seem to be using an extended world height - some of entries like minecraft:mossy_stone_brick_slab go up to index 380 (which is something like a height of 324). JER doesn't support this, and will raise an error like this one:
```
[Render thread/WARN] [jeresources/]: Error during loading of DIY data
java.lang.ArrayIndexOutOfBoundsException: Index 320 out of bounds for length 320
```
This isn't really something I can fix, as it's fundamentally a limitation of JER's format. I could in theory generate world-gen.json files for such worlds that would be valid, but that'd mean discarding all the data above y=256.

Interestingly enough, if I replicate your setup (including EMI), I don't get the error in question (I only do when I disable EMI). Probably EMI breaks JER's intergration and so it never ends up even trying to load the file.

**Comment by RundownRhino** [comment]
This is very similar to issue #11, except there the issue was that some worlds can go below -64, and here the issue is that some worlds can go above 256. Guess I can handle it the same way - discard all the data that JER can't show and print a warning about it.

**Comment by RundownRhino** [comment]
Should be fixed in 0.3.6, in the sense that the generated world-gen.json will omit heights above 255 and thereby be compatible with JER.

**Comment by RundownRhino** [comment]
> It also misses some modded ores like Immersive engineering ores in the worldgen file.

This seems like a different issue - please elaborate if you're still experiencing it. I don't stop anything missing on a cursory glance.
